mesonbuild/modules/java.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2021 The Meson development team
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pathlib
+from mesonbuild.mesonlib.universal import File
+from mesonbuild.build import CustomTarget
+from mesonbuild.optinterpreter import permitted_kwargs
+import os
+from mesonbuild.interpreterbase.decorators import FeatureNew, typed_pos_args
+import typing as T
+from . import ExtensionModule, ModuleReturnValue, ModuleState
+from ..interpreter import Interpreter
+from mesonbuild.interpreter.interpreterobjects import FileHolder
+
+class JavaModule(ExtensionModule):
+    @FeatureNew('Java Module', '0.59.0')
+    def __init__(self, interpreter: Interpreter):
+        super().__init__(interpreter)
+        self.methods.update({
+            'generate_native_header': self.generate_native_header,
+        })
+
+    @typed_pos_args('generate_native_header', (str, FileHolder), min_varargs=1, max_varargs=1)
```

Comment:
@dcbaker would appreciate reviewing this specifically.

Reply:
You don't need the min_ and max_ varargs here, your function only uses fixed positional arguments, varargs (variadic arguments) are for functions like `file()` which takes 0 or more strings and returns the an array with the same number of File elements. I have no idea what happens if you set those flags and don't set `vaargs=...`, you could be running into an odd corner of that.